Clarify default enabled services

cpaelzer opened the pull request:
NO-LP NO-JIRA NO-GH

## Proposed Commit Message
```
docs: Better describe control of auto-enabled services

Neither do we mention the dashboard while getting the
token nor the ability to not auto-enable services.
Users asking for it made me realize that this is a weak
spot in the documentation.
```

## Test Steps

- Look at the docs as if you'd know nothing and try to understand how to get a token and how to control the services that are enabled by default when attaching.
- No functional test

## Checklist

 - [n/a] I have updated or added any unit tests accordingly
 - [n/a] I have updated or added any integration tests accordingly
 - [x] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

cpaelzer commented:
@s-makin this is the first time I add an image, it LGTM on the [GH rendering](https://github.com/cpaelzer/ubuntu-pro-client/blob/clarify-default-enabled-services/docs/howtoguides/get_token_and_attach.md), will this pass well through spinx and RTD?

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -35,3 +46,24 @@ Enable services with: pro enable <service>
+controlled on the [Ubuntu Pro Dashboard](https://ubuntu.com/pro/dashboard).
+
+Recommended services are auto-enabled when attaching a system. Which services
+are considered recommended can, for some subscriptions, be controlled on the
```
I'm not sure about this bit here. The wording implies that you're toggling which services are recommended, but I was under the impression there are certain services that we recommend (which are auto-enabled) and the toggle just allows you to enable/disable those. If the latter is the case, let me know and I'll tighten up the wording.

cpaelzer replied:
The initial setting is what "we recommend", but you can toggle it to make your own selection.
Indeed the use of the word recommended is misleading and I only used it because that is how they were called before.

So the flow is 1. you setup a subscription -> 2. you go to dashboard, recommended services are selected by default -> 3. you can adjust there as/if you want -> 4. you attach and those enabled will be auto enabled.

s-makin replied:
That's what I thought - I'll recommend an alternative

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -35,3 +46,24 @@ Enable services with: pro enable <service>
+![Toggling recommended services in the Pro Dashboard](pro-dashboard-service-toggles.png)
+
+In case you can not control your subscription (depends on the subscription type
+and your users permissions related to it). Or if you do not want to change the
```
```suggestion
and the user permissions granted by it). Or if you do not want to change the
```

cpaelzer replied:
Umm, no :-)

Subscriptions differ and in them then users differ:
- free one - just defaults no permission to change via the web UI
- real subscription has some user management, you might have the permission to change things or only to see them

Your suggestion implies that the subscription would grant any permissions. That is not what I was saying. I wanted to avoid 500 more words but make it clear that (depending on complex conditions) you might or might not be able to see or tweak them.

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -35,3 +46,24 @@ Enable services with: pro enable <service>
+
+In case you can not control your subscription (depends on the subscription type
+and your users permissions related to it). Or if you do not want to change the
+defaults, but you want to not auto-enable anything while attaching a particular
```
```suggestion
defaults, but you do not want to auto-enable anything while attaching a particular
```

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -35,3 +46,24 @@ Enable services with: pro enable <service>
+and your users permissions related to it). Or if you do not want to change the
+defaults, but you want to not auto-enable anything while attaching a particular
+machine you can always pass `--no-auto-enable` to attach, but not auto-enable
+any of the services.
```
```suggestion
any services using the following command:
```

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -35,3 +46,24 @@ Enable services with: pro enable <service>
+In case you can not control your subscription (depends on the subscription type
+and your users permissions related to it). Or if you do not want to change the
+defaults, but you want to not auto-enable anything while attaching a particular
+machine you can always pass `--no-auto-enable` to attach, but not auto-enable
```
```suggestion
machine you can always pass `--no-auto-enable` to `attach`, but not auto-enable
```

s-makin commented:
> @s-makin this is the first time I add an image, it LGTM on the [GH rendering](https://github.com/cpaelzer/ubuntu-pro-client/blob/clarify-default-enabled-services/docs/howtoguides/get_token_and_attach.md), will this pass well through spinx and RTD?

I tested it and it looks fine to me, rendered with no problems. I wonder if it wouldn't look better centered on the page though? (Also, thank you for using alt-text :D)

cpaelzer commented:
> I wonder if it wouldn't look better centered on the page though?

That would go deep into inline html or at least html/css attributes.
I'm sure that will make compatible rendering to other formats not easier so I avoided that.

s-makin commented:
> > I wonder if it wouldn't look better centered on the page though?
> 
> That would go deep into inline html or at least html/css attributes. I'm sure that will make compatible rendering to other formats not easier so I avoided that.

Fair enough - let's leave it as-is for now then

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -35,3 +46,24 @@ Enable services with: pro enable <service>
+Your subscription controls which services are available to you as shown and
+controlled on the [Ubuntu Pro Dashboard](https://ubuntu.com/pro/dashboard).
+
+Recommended services are auto-enabled when attaching a system. Which services
+are considered recommended can, for some subscriptions, be controlled on the
+[Ubuntu Pro Dashboard](https://ubuntu.com/pro/dashboard) by toggling those.
```
```suggestion
Recommended services are auto-enabled by default when attaching a system.
You can choose which of the available services will be automatically
enabled or disabled when you attach by toggling them in the
[Ubuntu Pro Dashboard](https://ubuntu.com/pro/dashboard). 
Available services can always be enabled or disabled on the command line
with `pro enable` and `pro disable` after attaching.
```

s-makin replied:
Since you mentioned in the previous paragraph that the subscription controls which services are available, I think this wording is perhaps a bit clearer (and hopefully not too long)

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -35,3 +46,24 @@ Enable services with: pro enable <service>
+
+## Control of auto-enabled services
+
+Your subscription controls which services are available to you as shown and
+controlled on the [Ubuntu Pro Dashboard](https://ubuntu.com/pro/dashboard).
```
```suggestion
Your subscription controls which services are available to you and which ones
you can manage via the [Ubuntu Pro Dashboard](https://ubuntu.com/pro/dashboard).
```

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -35,3 +46,24 @@ Enable services with: pro enable <service>
+
+![Toggling recommended services in the Pro Dashboard](pro-dashboard-service-toggles.png)
+
+In case you can not control your subscription (depends on the subscription type
+and your users permissions related to it). Or if you do not want to change the
+defaults, but you do not want to auto-enable anything while attaching a particular
+machine you can always pass `--no-auto-enable` to `attach`, but not auto-enable
+any services using the following command:
```
```suggestion
If your subscription does not permit you to change the default
enabled services via the Dashboard, or if you want to keep the
defaults but do not want to auto-enable any services while attaching a particular
machine, you can pass the `--no-auto-enable` flag to `attach` using the following command:
```

s-makin reviewed (commented):
I've taken your previous advice on board and made some more holistic changes that I think should make the section clearer without adding too much bulk to the text. Please let me know though if I've added any other inaccuracies that need fixing.

cpaelzer commented:
> I've taken your previous advice on board and made some more holistic changes that I think should make the section clearer without adding too much bulk to the text. Please let me know though if I've added any other inaccuracies that need fixing.

Thanks to our discussion how/why this works all your v2 suggestions were fine and I added them. Thanks @s-makin !

cpaelzer commented:
Now that iteration 1 and 2 are done I think this should be good.
Opening up for a re-review by @s-makin and also adding @orndorffgrant which I have discussed this topic with as a technical pair of eyes before we merge.

s-makin commented on docs/howtoguides/get_token_and_attach.md:
```diff
@@ -1,10 +1,21 @@
+
 Retrieve your Ubuntu Pro token from the
-[Ubuntu Pro portal](https://ubuntu.com/pro/). You will log in with your "Single 
+[Ubuntu Pro portal](https://ubuntu.com/pro/). You will log in with your "Single
```
```suggestion
[Ubuntu Pro portal](https://ubuntu.com/pro/). Log in with your "Single
```

s-makin reviewed (approved):
LGTM :) just that one little nit to make it sound a bit more neutral
